lib/emoji.rb

```diff
@@ -63,19 +63,24 @@ def edit_emoji(emoji)
 
   # Public: Find an emoji by its aliased name. Return nil if missing.
   def find_by_alias(name)
-    names_index[name]
+    @mutex.synchronize do
```

Comment:
That's a good point. In that case, can we simply call `all` from `names_index`/`unicode_index` instead of having the `all unless defined @all` expression?

Alternatively to this whole PR, do you think it would be worth it to populate all emoji in memory as soon as this library loads? Then there would be no multithreading issues, and threads wouldn't be slowed down by the fact they need to be synchronized around `Emoji.find_by_*` calls.